lib/graphql/input_object_type.rb

```diff
@@ -27,6 +27,32 @@ def valid_non_null_input?(input)
       input_fields.all? { |name, field| field.type.valid_input?(input[name]) }
   end
 
+  def validate_non_null_input(input)
+    # Inputs are only required to provide an `all?` method and a `[]` method. That's not enough
+    # to get detailed errors, so if that's all we got, we can only provide generalized error.
+    # (see spec/support/mimimum_input_object.rb)
+    unless input.is_a?(Enumerable)
+      result = GraphQL::Query::InputValidationResult.new
+      result.add_problem("Invalid input was provided") unless valid_non_null_input?(input)
+      return result
+    end
+
+    result = GraphQL::Query::InputValidationResult.new
+
+    # Items in the input that are unexpected
+    input.reject { |name, value| input_fields[name] }.each do |name, value|
```

Comment:
This _should_ be handled by static validation... oh, but I guess if you're making an input object out of query variables, it would be nice to get some feedback?